bin/arg.ml

```diff
@@ -25,3 +25,98 @@ let path = Path.conv
 
 [@@@ocaml.warning "-32"]
 let file = path
+
+module Dep : sig
+  type t = Dune_file.Dep_conf.t
+
+  val conv : t conv
+
+  val file : string -> t
+  val alias : string -> t
+  val alias_rec : string -> t
+
+  val to_string_maybe_quoted : t -> string
+end = struct
+  type t = Dune_file.Dep_conf.t
+
+  let file s =
+    Dune_file.Dep_conf.File (String_with_vars.make_text Loc.none s)
+
+  let alias s =
+    Dune_file.Dep_conf.Alias (String_with_vars.make_text Loc.none s)
+
+  let alias_rec s =
+    Dune_file.Dep_conf.Alias_rec (String_with_vars.make_text Loc.none s)
+
+  let parse_alias s =
+    if not (String.is_prefix s ~prefix:"@") then
+      None
+    else
+      let pos, recursive =
+        if String.length s >= 2 && s.[1] = '@' then
+          (2, false)
+        else
+          (1, true)
+      in
+      let s = String.drop s pos in
+      Some (recursive, s)
+
+  let parser s =
+    let p =
+      Dune.Syntax.set Dune.Stanza.syntax Dune.Stanza.latest_version
+        Dune.Dune_file.Dep_conf.decode
+    in
+    match
+      Dune_lang.Decoder.parse p Univ_map.empty
+        (Dune_lang.parse_string ~fname:"command line"
+           ~mode:Dune_lang.Parser.Mode.Single s)
+    with
+    | Dune_file.Dep_conf.File sv as x ->
+      begin match String_with_vars.text_only sv with
+      | None -> `Ok x
+      | Some s ->
+        let x =
+          match parse_alias s with
+          | None -> x
+          | Some (true, s) -> alias_rec s
+          | Some (false, s) -> alias s
```

Comment:
I believe this function is parsing `(file @foo)` as `Alias_rec "foo"`. We should probably call `parse_alias` before

Reply:
You are right, I had missed that; will fix.

Reply:
This is fixed now, thanks.